### ISSUE TITLE
Consider other power plants in consumption datapoint

### ISSUE THREAD

**Issue by theknut** [bolliy/ioBroker.sun2000#76]
### What would you like to be added?

Allow the output of other power plants to be used in the calculation of `collected.houseConsumption`.

### Why is this needed?

The `houseConsumption` state is always wrong if other power plants, such as balcony power plants are used. E.g. in my case the inverters are directly in line with the huawei smart meter and therefore the smart meter thinks there is no consumption because the other power plant is already covering the entire house consumption.

**Comment by theknut** [comment]
Thanks for the quick response.
Even though I respect your decision, I still think it's part of the
adapter. Otherwise the adapter should not even try to calculate the house
consumption if not all data is accessible to calculate it. I think
especially in Germany this is quite a common use case. First you buy a BKW,
then you get hooked and install a bigger power plant. Almost all of my
friends are operating a big power plant and at least one BKW or mini PV.

Am Do., 11. Apr. 2024 um 17:07 Uhr schrieb bolliy ***@***.***

**Comment by bolliy** [comment]
Of course you're right, but Huawei doesn't take other power plants in the FusionSolar Portal or App. I'm just afraid that more and more requirements of this kind. I'll put your issue on hold. Maybe I'll change my view

**Comment by MM1834** [comment]
I support Stephan's decision as it would become very complex to cover all possible situation on how to integrate other power sources. In my case I have some micro inverters but also a balcony solar systems with integrated battery. I only measure the net output of that sources.

Huawei Fusion supports other power sources even the visualization is very poor and there seems no good documentation available. As long as you have only one SUN2000 and modbus 1 is available, just install a 2nd huawei power sensor tht measures all external sources (net output into grid).

sun2000- houseConsumtion: The calculation is not valid in such a case. Today the values shows 0.434 kWh (!!!) even today's consumption is 19.11 according to fusion (what seems to be realistic to me)

**Comment by theknut** [comment]
Fair point concerning Fusion Solar. It's wrong there as well. Since you
already have datapoints which consider efficiency such as
`collected.inputPower` and `collected.inputPowerWithEfficiencyLoss` you
could add a separate datapoint `collected.houseConsumptionCalculated`.

Am Do., 11. Apr. 2024 um 17:32 Uhr schrieb bolliy ***@***.***

**Comment by theknut** [comment]
The houseConsumption datapoint is the current consumption by the house.

I don't think it's vers complicated. Just calculate all other sources into
one datapoint and then the adapter allows to provide one datapoint which is
subtracted. It's really not complicated at all.

MM1834 ***@***.***> schrieb am Do., 11. Apr. 2024, 20:07:

**Comment by bolliy** [comment]
@theknut I have implemented your issue in version 1.4.0 
Please test and close this if necessary.

Stephan

**Comment by theknut** [comment]
How can I use it? Just setting externalPower to the output of the BKW?

**Comment by bolliy** [comment]
You can enter the current power output of BKW in the object state **sun2000.0.control.externalPower**

**Comment by bolliy** [comment]
I think it was implemented
